src/Microsoft.DotNet.Build.Tasks/PackageFiles/packageresolve.targets

```diff
@@ -108,8 +108,20 @@
       <_ReferenceCopyLocalPathsFromPackage Remove="@(_ReferenceCopyLocalPathsFileNamesToRemove)" />
 
       <!-- add the filtered resolved package output -->
-      <Reference Include="@(_ReferenceFromPackage)" />
+      <Reference Condition="'$(SourceIndex)' != 'true'" Include="@(_ReferenceFromPackage)" />
```

Comment:
I'd prefer it if you did this as a separate target.  Looks to me like it it is all just modifications that could be done in a Target that runs AfterTargets=ResolveNuGetPackages.  For that matter this doesn't even need to go into buildtools.